app/org/sagebionetworks/bridge/play/controllers/AuthenticationController.java

```diff
@@ -93,9 +117,8 @@ public Result resetPassword() throws Exception {
     private Result signInWithRetry(int retryCounter) throws Exception {
         UserSession session = getSessionIfItExists();
         if (session == null) {
-            JsonNode json = requestToJSON(request());
             SignIn signIn = parseJson(request(), SignIn.class);
-            Study study = getStudyOrThrowException(json);
+            Study study = studyService.getStudy(signIn.getStudyId());
```

Comment:
I added a couple of tests for this because it doesn't seem to be verified working anywhere (at the controller level). Then fixed it so this is thrown correctly on sign in and email sign in.